magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merge_stereo.py

```diff
@@ -0,0 +1,141 @@
+    """
+
+    process_name = source
+    stereo_DL1_dir = f"{target_dir}/v{__version__}/{source}"
+    listOfNightsLST = np.sort(glob.glob(f"{stereo_DL1_dir}/DL1Stereo/*"))
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+    for nightLST in listOfNightsLST:
+        night = nightLST.split("/")[-1]
+        stereoMergeDir = f"{stereo_DL1_dir}/DL1Stereo/{night}/Merged"
+        os.makedirs(f"{stereoMergeDir}/logs", exist_ok=True)
+
+        if len(glob.glob(f"{nightLST}/dl1_stereo*.h5")) < 1:
+            continue
+
+        slurm = slurm_lines(
+            queue="short",
+            job_name=f"{process_name}_stereo_merge",
+            out_name=f"{stereoMergeDir}/logs/slurm-%x.%A_%a",
+        )
+        rc = rc_lines(
+            store=f"{nightLST} ${{SLURM_JOB_ID}}", out=f"{stereoMergeDir}/logs/list"
+        )
+        os.system(f"echo {nightLST} >> {stereoMergeDir}/logs/list_dl0.txt")
+        lines = (
+            slurm
+            + [
+                f"conda run -n {env_name} merge_hdf_files --input-dir {nightLST} --output-dir {stereoMergeDir} --run-wise >{stereoMergeDir}/logs/merge_{night}_${{SLURM_JOB_ID}}.log 2>&1\n"
+            ]
+            + rc
+        )
+
+        with open(f"{source}_StereoMerge_{night}.sh", "w") as f:
+            f.writelines(lines)
+
+
+def main():
+    """
+    Here we read the config_general.yaml file and call the functions defined above.
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config-file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./config_general.yaml",
+        help="Path to a configuration file",
+    )
+
+    args = parser.parse_args()
+    with open(
+        args.config_file, "rb"
+    ) as f:  # "rb" mode opens the file in binary format for reading
+        config = yaml.safe_load(f)
+
+    target_dir = Path(config["directories"]["workspace_dir"])
+
+    env_name = config["general"]["env_name"]
+
+    source_in = config["data_selection"]["source_name_database"]
+    source = config["data_selection"]["source_name_output"]
+    cluster = config["general"]["cluster"]
+
+    source_list = []
+
+    if source_in is None:
+        source_list = joblib.load("list_sources.dat")
+
+    else:
+        source_list.append(source)
+
+    for source_name in source_list:
+
+        print("***** Merging DL1Stereo files run-wise...")
+        MergeStereo(target_dir, env_name, source_name, cluster)
+
+        list_of_merge = glob.glob(f"{source_name}_StereoMerge_*.sh")
+        if len(list_of_merge) < 1:
+            print("Warning: no bash script has been produced")
+            continue
+
+        launch_jobs = ""
+
+        for n, run in enumerate(list_of_merge):
+            launch_jobs += (
+                " && " if n > 0 else ""
+            ) + f"merge{n}=$(sbatch --parsable {run})"
```

Comment:
you do not use this shell variable 'merge', right? if so you can simplify the command and you probably do not need --parsable